### DISCUSSION TITLE
{{Today}} token in Results Template is there one?

### DISCUSSION THREAD

**Discussion by LeonArmston** [microsoft-search/pnp-modern-search#3697] [Q&A] [answered]
In my custom search results template I would like to show an expiry date for a document in red if the date is less than today i.e. expired.

I'd like to use a a handlebar condition to handle this i.e. If (conditional) but it appears there is no Today variable or helper.

Is there something I'm missing or would this be a good feature for PnP Search?

**Comment by kasperbolarsen** [comment]
These doesn't work ? https://microsoft-search.github.io/pnp-modern-search/usage/search-results/tokens/#date-tokens

**Reply by LeonArmston** [reply]
Hi @kasperbolarsen 

I think these work in the  query template i.e. DueDate:{{CurrentDate}} AND Path:"https://pnp.sharepoint.com"

But dont appear to work in the Results Template Handlbars/HTML

![image](https://github.com/microsoft-search/pnp-modern-search/assets/12968962/b908f9d0-ec08-41c7-ad4d-a2019bdb8f92)

Cant seem to use any date tokens or see them in Debug.

Would be a handy feature I'd imagine.

**Comment by patrikhellgren** [comment, marked as answer]
Here is one way of doing that in a column handlebars template but it could of course be used in a full layout also. Unfortunately the {{moment}} helper is not very well documented but with the moment.js documentation and some trial and error you will get pretty far.

``` html
<style>
  .expiredDate {
      color: red
  };
  .normalDate {
      color: black
  };
</style>
{{#if (moment (slot item @root.slots.Date) isBefore=(moment))}}
  <div class='expiredDate'>
    {{getDate (slot item @root.slots.Date) 'LL'}}
  </div>
{{else}}
  <div class='normalDate'>
    {{getDate (slot item @root.slots.Date) 'LL'}}
  </div>
{{/if}}
```

**Reply by LeonArmston** [reply]
@patrikhellgren - that works so well!! Thanks so much!!! 

The {{moment}} helper looks very helpful and a good one to now know!!

Thanks again Sir!
